### ISSUE TITLE
Zod 4.5: `consistent-schema-var-name` reports `z.validate()` results as schemas

### ISSUE THREAD

**Issue by marcalexiei** [marcalexiei/eslint-plugin-zod#415]
**📦 Affected package:** `eslint-plugin-zod` (`eslint-plugin-zod-mini` too — the
form only takes one)

## 🧩 Description

zod `4.5.0` adds `z.validate(schema, input)` / `z.validateAsync(…)`, which return `boolean` and `Promise<boolean>`. 
`consistent-schema-var-name` treats them as schema factories and asks for a `Schema` suffix on the result variable.

Neither name is in `ZOD_NON_SCHEMA_PRODUCING_METHODS`, 
the list the rule uses to bail out. 

## 🪜 How to reproduce

```ts
// eslint.config.js → { rules: { 'zod/consistent-schema-var-name': ['error', { after: 'Schema' }] } }
import * as z from 'zod';

const userSchema = z.object({ name: z.string() });

const isValid = z.validate(userSchema, input);
//    ^ Rename this Zod schema to "isValidSchema"

const isValidLater = z.validateAsync(userSchema, input);
//    ^ Rename this Zod schema to "isValidLaterSchema"
```

## ✅ Expected behavior

No report. 
`isValid` is a `boolean`, not a schema, so the naming convention does not apply.
